selfdrive/updated.py

```diff
@@ -315,46 +287,63 @@ def attempt_update():
 
 
 def main():
-  update_failed_count = 0
-  overlay_init_done = False
   params = Params()
 
   if params.get("DisableUpdates") == b"1":
     raise RuntimeError("updates are disabled by param")
 
-  if not os.geteuid() == 0:
+  if os.geteuid() != 0:
     raise RuntimeError("updated must be launched as root!")
 
   # Set low io priority
   p = psutil.Process()
   if psutil.LINUX:
     p.ionice(psutil.IOPRIO_CLASS_BE, value=7)
 
-  ov_lock_fd = open('/tmp/safe_staging_overlay.lock', 'w')
+  ov_lock_fd = open(LOCK_FILE, 'w')
   try:
     fcntl.flock(ov_lock_fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
   except IOError:
     raise RuntimeError("couldn't get overlay lock; is another updated running?")
 
-  # Wait a short time before our first update attempt
-  # Avoids race with IsOffroad not being set, reduces manager startup load
-  time.sleep(30)
-  wait_helper = WaitTimeHelper()
+  # Setup a signal handler to immediately trigger an update
```

Comment:
What was wrong with the class? Now there is all this random code in the middle of the logic.

Reply:
though the class was pretty confusing when first reading updated since it was basically a wrapper around threading.Event, can revert this

Reply:
reverted